game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java

```diff
@@ -33,7 +36,7 @@
  * until a new game has been started (TODO: check if the lobby
  * uses mainpanel at all)
  */
-public class MainPanel extends JPanel implements Observer {
+public class MainPanel extends JPanel implements Observer, ScreenChangeListener {
```

Comment:
I'm curious because the other class no longer extends Observable: Is implementing the Observer interface still necessary?

Reply:
In short yes. `ScreenChangeListener` is the new interface instead of Observer relative to the other class. There is still more work needed to remove Observer here. I tried to remove it everywhere and it became a tangle... So just removing the one example carefully in this iteration. 

For specific context, we had 3 observer usages in `MainPanel`, chipping away at this we would be down to two. The other two observer usages are:
1. player selection (to enable play button), main panel observes the `ISetupPanel`
2. changing game selection, main panel observes `GameSelectorModel`

One important thing to notice is in the update the observer/observee relationship is defined outside of MainPanel. That is a more fundamental design shift, important for future refactoring which can take advantage of that decoupling.